__exekutir__/roles/kommandir_workspace_setup/tasks/main.yml

```diff
@@ -123,8 +123,9 @@
   git:
     repo: "{{ item.repo }}"
     dest: "{{ kommandir_workspace }}/cache/{{ item.cachepath }}"
-    # Shallow clone is much faster, two commits will cover a merge-commit
-    depth: "{{ item.depth | default(2) }}"
+    # Shallow clone is much faster, two commits will cover a merge-commit and will be set per repo.
+    # default(omit) to workaround failure with python-bugzilla repo
+    depth: "{{ item.depth | default(omit) }}"
```

Comment:
Interesting idea, and your motivations are good.  The only way I know of is by hacking in jinja2 templates.  Though I wonder if what you have would work (I've not tried it).  Something like this would do what you're suggesting, but as you can see, it's a bit fugly (why I like to avoid it):

```yaml
...
- name: Kommandir's cache content checked out from git when git_cache_args non-empty, result is false, and both kommandir_workspace_dirty flags set
  git:
    repo: "{{ item.repo }}"
    dest: "{{ kommandir_workspace }}/cache/{{ item.cachepath }}"
    # Shallow clone is much faster, two commits will cover a merge-commit
    depth: |
        {%- if item.version is defined and item.version not in [None,'',[],{}] -%}\
            {{ omit }}\
        {%- else -%}\
            {{ item.depth | default(2) }}\
        {%- endif -%}
    # HEAD is the default if not specified
    version: "{{ item.version | default(omit) }}"
    recursive: "{{ item.recursive | default(False) }}"
  when: kommandir_workspace_dirty and hostvars.exekutir.kommandir_workspace_dirty and
        not result and git_cache_args is defined
  with_items: '{{ git_cache_args }}'
  ...
```

My experience with Ansible is, the more you try and stuff it into the "programming language" box, the more it will cause general misery / frustration.  Another way is to write the above into an actual jinja2 template file, and then read it in using ``{{ lookup('template','path/to/file') }}``.  The down-side is it means the future engineer has to go hunting for that file to understand what the task does.

I'm totally open to fixing this in the spirit of what you suggest.  Just ignorant of a good way to do it.  Also, to be sure, there are probably many more areas where a similar maintainability-advice could be applied.  I'd like to find/fix all of them if we can, just not sure of what's the best/better way.  Can I open an issue to remind me/we/us to play with it, and reference your example/suggestion?